diff --git a/node_modules/exif-js/exif.d.ts b/node_modules/exif-js/exif.d.ts
index 31a6fb0..a4be514 100644
--- a/node_modules/exif-js/exif.d.ts
+++ b/node_modules/exif-js/exif.d.ts
@@ -1,5 +1,20 @@
+/* eslint-disable */
+type ExifData = {
+  Orientation: number
+}
+
+declare global {
+  interface HTMLImageElement {
+    exifdata?: ExifData
+  }
+}
+
+type SourceImage = HTMLImageElement | HTMLInputElement | string
+
+type Callback = () => void
+
 interface EXIFStatic {
-    getData(url: string, callback: any): any;
+    getData(source: SourceImage, callback: Callback): void;
     getTag(img: any, tag: any): any;
     getAllTags(img: any): any;
     pretty(img: any): string;
