diff -ruN cpuinfo/BUILD.bazel ../external-patched/cpuinfo/BUILD.bazel
--- cpuinfo/BUILD.bazel	2024-02-21 13:53:11.458549589 +0000
+++ ../external-patched/cpuinfo/BUILD.bazel	2024-02-21 13:17:53.047760336 +0000
@@ -106,6 +106,7 @@
     name = "cpuinfo_impl",
     srcs = select({
         ":linux_x86_64": COMMON_SRCS + X86_SRCS + LINUX_SRCS + LINUX_X86_SRCS,
+        ":linux_k8": COMMON_SRCS + X86_SRCS + LINUX_SRCS + LINUX_X86_SRCS,
         ":linux_arm": COMMON_SRCS + ARM_SRCS + LINUX_SRCS + LINUX_ARM32_SRCS,
         ":linux_armhf": COMMON_SRCS + ARM_SRCS + LINUX_SRCS + LINUX_ARM32_SRCS,
         ":linux_armv7a": COMMON_SRCS + ARM_SRCS + LINUX_SRCS + LINUX_ARM32_SRCS,
@@ -196,7 +197,7 @@
 
 config_setting(
     name = "linux_x86_64",
-    values = {"cpu": "k8"},
+    values = {"cpu": "x86_64"},
 )
 
 config_setting(
@@ -240,6 +241,11 @@
 )
 
 config_setting(
+    name = "linux_k8",
+    values = {"cpu": "k8"},
+)
+
+config_setting(
     name = "macos_x86_64",
     values = {
         "apple_platform_type": "macos",
