diff --git a/oneTBB-2021.11.0/BUILD.bazel b/oneTBB-2021.11.0-patched/BUILD.bazel
index 3881d68..c67adb5 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -16,6 +16,8 @@
 # use Bazel internally. The Bazel build can have security risks or
 # optimization gaps.
 
+load("@bazel_skylib//lib:selects.bzl", "selects")
+
 package(
     default_visibility = ["//visibility:public"],
 )
@@ -35,10 +37,7 @@ cc_library(
         "include/oneapi/tbb/*.h",
         "include/oneapi/tbb/detail/*.h",
     ]),
-    copts = ["-w"] + select({
-        "@platforms//os:windows": [""],
-        "//conditions:default": ["-mwaitpkg"],
-    }),
+    copts = ["-w", "-mwaitpkg", "-mrtm"],
     defines =
         select({
             "@platforms//cpu:x86_64": ["__TBB_NO_IMPLICIT_LINKAGE"],
@@ -46,8 +45,8 @@ cc_library(
                 "USE_PTHREAD",
             ],
         }) +
-        select({
-            "@platforms//os:osx": ["_XOPEN_SOURCE"],
+        selects.with_or({
+            ("@platforms//os:ios", "@platforms//os:osx"): ["_XOPEN_SOURCE"],
             "//conditions:default": [],
         }),
     includes = [
