diff --git a/google/cloud/storage/BUILD b/google/cloud/storage/BUILD
index 1ae92ed1e..b2397204a 100644
--- a/google/cloud/storage/BUILD
+++ b/google/cloud/storage/BUILD
@@ -55,7 +55,7 @@ cc_library(
         "@boringssl//:ssl",
         "@com_github_curl_curl//:curl",
         "@com_github_google_crc32c//:crc32c",
-        "@com_github_nlohmann_json//:nlohmann_json",
+        "@json//:json", # Niantic's internal use of nlohmann json uses this repo name and target name
         "@com_google_absl//absl/algorithm:container",
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/time",
@@ -82,7 +82,7 @@ cc_library(
         "@boringssl//:ssl",
         "@com_github_curl_curl//:curl",
         "@com_github_google_crc32c//:crc32c",
-        "@com_github_nlohmann_json//:nlohmann_json",
+        "@json//:json", # Niantic's internal use of nlohmann json uses this repo name and target name
         "@com_google_absl//absl/memory",
         "@com_google_absl//absl/strings:str_format",
         "@com_google_absl//absl/time",
