diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3987eac..cde8e17 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,12 @@ include(CheckCXXSymbolExists)

 project(FlatBuffers)

+# ROR_DEPS_EDIT(1/28/19): use NEW cmake policy for CMP0077, basically not overwriting normal variables
+# (set by set or -D at command line) if they haven't been cached.
+if (POLICY CMP0077)
+  cmake_policy(SET CMP0077 NEW)
+endif()
+
 # NOTE: Code coverage only works on Linux & OSX.
 option(FLATBUFFERS_CODE_COVERAGE "Enable the code coverage build option." OFF)
 option(FLATBUFFERS_BUILD_TESTS "Enable the build of tests and samples." ON)
@@ -213,7 +219,6 @@ if(EXISTS "${CMAKE_TOOLCHAIN_FILE}")
 elseif(APPLE)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -stdlib=libc++")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -pedantic -Werror -Wextra -Wno-unused-parameter")
-  set(FLATBUFFERS_PRIVATE_CXX_FLAGS "-Wold-style-cast")
 elseif(CMAKE_COMPILER_IS_GNUCXX)
   if(CYGWIN)
     set(CMAKE_CXX_FLAGS
@@ -224,7 +229,6 @@ elseif(CMAKE_COMPILER_IS_GNUCXX)
   endif(CYGWIN)
   set(CMAKE_CXX_FLAGS
     "${CMAKE_CXX_FLAGS} -Wall -pedantic -Werror -Wextra -Werror=shadow")
-  set(FLATBUFFERS_PRIVATE_CXX_FLAGS "-Wold-style-cast")
   if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 4.4)
     if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 7.0)
       set(CMAKE_CXX_FLAGS
@@ -242,7 +246,6 @@ elseif(CMAKE_COMPILER_IS_GNUCXX)
 elseif(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
   set(CMAKE_CXX_FLAGS
       "${CMAKE_CXX_FLAGS} -std=c++0x -Wall -pedantic -Werror -Wextra -Wno-unused-parameter")
-  set(FLATBUFFERS_PRIVATE_CXX_FLAGS "-Wold-style-cast")
   if(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.8)
     list(APPEND FLATBUFFERS_PRIVATE_CXX_FLAGS "-Wimplicit-fallthrough" "-Wextra-semi" "-Werror=unused-private-field") # enable warning
   endif()
@@ -519,7 +522,7 @@ include(CMake/Version.cmake)
 if(FLATBUFFERS_INSTALL)
   include(GNUInstallDirs)

-  install(DIRECTORY include/flatbuffers DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+  install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})

   set(FB_CMAKE_DIR "${CMAKE_INSTALL_LIBDIR}/cmake/flatbuffers")

diff --git a/android/jni/Application.mk b/android/jni/Application.mk
index ca9e800..710bc30 100644
--- a/android/jni/Application.mk
+++ b/android/jni/Application.mk
@@ -16,5 +16,5 @@
 APP_PLATFORM := android-9
 APP_PROJECT_PATH := $(call my-dir)/..
 APP_STL ?= stlport_static
-APP_ABI := armeabi-v7a
+APP_ABI := armeabi-v7a arm64-v8a
 APP_CPPFLAGS += -std=c++11
diff --git a/include/flatbuffers/base.h b/include/flatbuffers/base.h
index 9557380..40eb366 100644
--- a/include/flatbuffers/base.h
+++ b/include/flatbuffers/base.h
@@ -374,6 +374,13 @@ T ReadScalar(const void *p) {
   return EndianScalar(*reinterpret_cast<const T *>(p));
 }

+// added 5/27/2018 to enable sorted lookups of mutable data types
+template<typename T>
+__supress_ubsan__("alignment")
+T ReadMutableScalar(void *p) {
+  return EndianScalar(*reinterpret_cast<T *>(p));
+}
+
 template<typename T>
 // UBSAN: C++ aliasing type rules, see std::bit_cast<> for details.
 __supress_ubsan__("alignment")
diff --git a/include/flatbuffers/flatbuffers.h b/include/flatbuffers/flatbuffers.h
index c4dc5bc..9fc442d 100644
--- a/include/flatbuffers/flatbuffers.h
+++ b/include/flatbuffers/flatbuffers.h
@@ -104,6 +104,10 @@ template<typename T> struct IndirectHelper {
   static return_type Read(const uint8_t *p, uoffset_t i) {
     return EndianScalar((reinterpret_cast<const T *>(p))[i]);
   }
+  // added 5/27/2018 to enable sorted lookups of mutable data types
+  static mutable_return_type ReadMutable(uint8_t *p, uoffset_t i) {
+    return EndianScalar((reinterpret_cast<T *>(p))[i]);
+  }
 };
 template<typename T> struct IndirectHelper<Offset<T>> {
   typedef const T *return_type;
@@ -113,6 +117,12 @@ template<typename T> struct IndirectHelper<Offset<T>> {
     p += i * sizeof(uoffset_t);
     return reinterpret_cast<return_type>(p + ReadScalar<uoffset_t>(p));
   }
+  // added 5/27/2018 to enable sorted lookups of mutable data types
+  static mutable_return_type ReadMutable(uint8_t *p, uoffset_t i) {
+    p += i * sizeof(uoffset_t);
+    return reinterpret_cast<mutable_return_type>(
+        p + ReadMutableScalar<uoffset_t>(p));
+    }
 };
 template<typename T> struct IndirectHelper<const T *> {
   typedef const T *return_type;
@@ -121,6 +131,10 @@ template<typename T> struct IndirectHelper<const T *> {
   static return_type Read(const uint8_t *p, uoffset_t i) {
     return reinterpret_cast<const T *>(p + i * sizeof(T));
   }
+  // added 5/27/2018 to enable sorted lookups of mutable data types
+  static mutable_return_type ReadMutable(uint8_t *p, uoffset_t i) {
+    return reinterpret_cast<T *>(p + i * sizeof(T));
+  }
 };

 // An STL compatible iterator implementation for Vector below, effectively
@@ -132,7 +146,7 @@ template<typename T, typename IT> struct VectorIterator {
   typedef IT *pointer;
   typedef IT &reference;

-  VectorIterator(const uint8_t *data, uoffset_t i)
+  VectorIterator(uint8_t *data, uoffset_t i)
       : data_(data + IndirectHelper<T>::element_stride * i) {}
   VectorIterator(const VectorIterator &other) : data_(other.data_) {}
   VectorIterator() : data_(nullptr) {}
@@ -171,6 +185,11 @@ template<typename T, typename IT> struct VectorIterator {

   IT operator->() const { return IndirectHelper<T>::Read(data_, 0); }

+  // added 5/29/2018 to support mutable iterators
+  IT operator*() { return IndirectHelper<T>::ReadMutable(data_, 0); }
+
+  IT operator->() { return IndirectHelper<T>::ReadMutable(data_, 0); }
+
   VectorIterator &operator++() {
     data_ += IndirectHelper<T>::element_stride;
     return *this;
@@ -214,7 +233,7 @@ template<typename T, typename IT> struct VectorIterator {
   }

  private:
-  const uint8_t *data_;
+  uint8_t *data_;
 };

 template<typename Iterator>
@@ -283,6 +302,30 @@ template<typename T> class Vector {
     return reinterpret_cast<const void *>(Data() + o);
   }

+  // added 5/27/2018 to enable support for stl container find() semantics
+  template<typename K> iterator find(K key) {
+    void *search_result = std::bsearch(
+        &key, Data(), size(), IndirectHelper<T>::element_stride, KeyCompare<K>);
+
+    if (!search_result) {
+      return end();  // Key not found.
+    }
+
+    return iterator(static_cast<uint8_t *>(search_result), 0);
+  }
+
+  // added 5/27/2018 to enable support for stl container find() semantics
+  template<typename K> const_iterator find(K key) const {
+    void *search_result = std::bsearch(
+        &key, Data(), size(), IndirectHelper<T>::element_stride, KeyCompare<K>);
+
+    if (!search_result) {
+      return end();  // Key not found.
+    }
+
+    return const_iterator(static_cast<uint8_t *>(search_result), 0);
+  }
+
   iterator begin() { return iterator(Data(), 0); }
   const_iterator begin() const { return const_iterator(Data(), 0); }

@@ -331,8 +374,8 @@ template<typename T> class Vector {
   }

   // The raw data in little endian format. Use with care.
-  const uint8_t *Data() const {
-    return reinterpret_cast<const uint8_t *>(&length_ + 1);
+  uint8_t *Data() const {
+    return reinterpret_cast<uint8_t *>(&length_ + 1);
   }

   uint8_t *Data() { return reinterpret_cast<uint8_t *>(&length_ + 1); }
@@ -354,19 +397,22 @@ template<typename T> class Vector {
     return IndirectHelper<T>::Read(element, 0);
   }

- protected:
-  // This class is only used to access pre-existing data. Don't ever
-  // try to construct these manually.
-  Vector();
+  // added 5/27/2018 to enable sorted lookups of mutable data types
+  template<typename K>
+  mutable_return_type LookupMutableObjectByKey(K key) const {
+    void *search_result = std::bsearch(
+        &key, Data(), size(), IndirectHelper<T>::element_stride, KeyCompare<K>);
+    if (!search_result) {
+      return nullptr;  // Key not found.
+    }
+    uint8_t *element = reinterpret_cast<uint8_t *>(search_result);

-  uoffset_t length_;
+    return IndirectHelper<T>::ReadMutable(element, 0);
+  }

- private:
-  // This class is a pointer. Copying will therefore create an invalid object.
-  // Private and unimplemented copy constructor.
-  Vector(const Vector &);
   Vector &operator=(const Vector &);

+  // grayson added 6/29/2018 to enable support for external bsearch
   template<typename K> static int KeyCompare(const void *ap, const void *bp) {
     const K *key = reinterpret_cast<const K *>(ap);
     const uint8_t *data = reinterpret_cast<const uint8_t *>(bp);
@@ -376,6 +422,18 @@ template<typename T> class Vector {
     // result here.
     return -table->KeyCompareWithValue(*key);
   }
+
+  protected:
+   // This class is only used to access pre-existing data. Don't ever
+   // try to construct these manually.
+   Vector();
+
+   mutable uoffset_t length_;
+
+  private:
+   // This class is a pointer. Copying will therefore create an invalid object.
+   // Private and unimplemented copy constructor.
+   Vector(const Vector &);
 };

 // Represent a vector much like the template above, but in this case we
@@ -1855,7 +1913,7 @@ class FlatBufferBuilder {
     extern T Pack(const S &);
     typedef T (*Pack_t)(const S &);
     std::vector<T> vv(len);
-    std::transform(v, v + len, vv.begin(), static_cast<Pack_t &>(Pack));
+    std::transform(v, v + len, vv.begin(), (Pack_t*)&Pack);
     return CreateVectorOfSortedStructs<T>(vv, len);
   }

diff --git a/include/flatbuffers/reflection.h b/include/flatbuffers/reflection.h
index 052e6d9..b02cef0 100644
--- a/include/flatbuffers/reflection.h
+++ b/include/flatbuffers/reflection.h
@@ -39,8 +39,12 @@ inline bool IsInteger(reflection::BaseType t) {
 inline bool IsFloat(reflection::BaseType t) {
   return t == reflection::Float || t == reflection::Double;
 }
+// Modified on 11/04/2019
 inline bool IsLong(reflection::BaseType t) {
-  return t == reflection::Long || t == reflection::ULong;
+  return t == reflection::Long;
+}
+inline bool IsULong(reflection::BaseType t) {
+    return t == reflection::ULong;
 }

 // Size of a basic type, don't use with structs.
diff --git a/src/idl_gen_cpp.cpp b/src/idl_gen_cpp.cpp
index f78ee26..6ac24d9 100644
--- a/src/idl_gen_cpp.cpp
+++ b/src/idl_gen_cpp.cpp
@@ -1882,6 +1882,11 @@ class CppGenerator : public BaseGenerator {
           "    return static_cast<int>({{FIELD_NAME}}() > val) - "
           "static_cast<int>({{FIELD_NAME}}() < val);";
       code_ += "  }";
+
+      // Added on 5/28/2018 to be able to look up keys
+      code_ += "  {{KEY_TYPE}} Key() const {";
+      code_ += "    return {{FIELD_NAME}}();";
+      code_ += "  }";
     }
   }

diff --git a/src/idl_gen_cpp.cpp b/src/idl_gen_cpp.cpp
index f78ee26..0d82b55 100644
--- a/src/idl_gen_cpp.cpp
+++ b/src/idl_gen_cpp.cpp
@@ -2717,18 +2717,24 @@ class CppGenerator : public BaseGenerator {
     code_.SetValue("STRUCT_NAME", Name(struct_def));
     code_.SetValue("NATIVE_NAME",
                    NativeName(Name(struct_def), &struct_def, opts_));
-    auto native_name =
-        NativeName(WrapInNameSpace(struct_def), &struct_def, parser_.opts);
-    code_.SetValue("POINTER_TYPE",
-                   GenTypeNativePtr(native_name, nullptr, false));

     if (opts_.generate_object_based_api) {
       // Generate the X::UnPack() method.
       code_ +=
           "inline " + TableUnPackSignature(struct_def, false, opts_) + " {";

-      code_ +=
-          "  {{POINTER_TYPE}} _o = {{POINTER_TYPE}}(new {{NATIVE_NAME}}());";
+      if(opts_.g_cpp_std == cpp::CPP_STD_X0) {
+        auto native_name =
+            NativeName(WrapInNameSpace(struct_def), &struct_def, parser_.opts);
+        code_.SetValue("POINTER_TYPE",
+                       GenTypeNativePtr(native_name, nullptr, false));
+        code_ +=
+            "  {{POINTER_TYPE}} _o = {{POINTER_TYPE}}(new {{NATIVE_NAME}}());";
+      } else if(opts_.g_cpp_std == cpp::CPP_STD_11) {
+        code_ += "  auto _o = std::unique_ptr<{{NATIVE_NAME}}>(new {{NATIVE_NAME}}());";
+      } else {
+        code_ += "  auto _o = std::make_unique<{{NATIVE_NAME}}>();";
+      }
       code_ += "  UnPackTo(_o.get(), _resolver);";
       code_ += "  return _o.release();";

