diff --git a/node_modules/qrcode/lib/renderer/svg-tag.js b/node_modules/qrcode/lib/renderer/svg-tag.js
index d0839fe..3bffe51 100644
--- a/node_modules/qrcode/lib/renderer/svg-tag.js
+++ b/node_modules/qrcode/lib/renderer/svg-tag.js
@@ -65,6 +65,7 @@ exports.render = function render (qrData, options, cb) {
 
   const path =
     '<path ' + getColorAttrib(opts.color.dark, 'stroke') +
+    ' stroke-width="1.02"' + // This fixes tiny gaps between strokes on iOS
     ' d="' + qrToPath(data, size, opts.margin) + '"/>'
 
   const viewBox = 'viewBox="' + '0 0 ' + qrcodesize + ' ' + qrcodesize + '"'
