diff --git tensorflow/lite/delegates/gpu/delegate.cc tensorflow/lite/delegates/gpu/delegate.cc
index 63ed22373dd..87373b3fc78 100644
--- tensorflow/lite/delegates/gpu/delegate.cc
+++ tensorflow/lite/delegates/gpu/delegate.cc
@@ -117,7 +117,10 @@ class DelegateKernel {
   ~DelegateKernel() { --delegate_->num_delegate_kernels_; }
 
   absl::Status Prepare(TfLiteContext* context,
-                       const TfLiteDelegateParams* delegate_params) {
+                       const TfLiteDelegateParams* delegate_params,
+                       absl::Span<const uint8_t> serialized_kernel_cache,
+                       absl::Span<const uint8_t> serialized_model_cache,
+                       bool hold_model_cache) {
     thread_id_prepare_ = std::this_thread::get_id();
 
     // Extract TFLite delegate execution plan from the context and convert it
@@ -133,13 +136,17 @@ class DelegateKernel {
     const int experimental_flags = delegate_->options().experimental_flags;
     if (experimental_flags & TFLITE_GPU_EXPERIMENTAL_FLAGS_CL_ONLY) {
       RETURN_IF_ERROR(
-          InitializeOpenClApi(&graph, &builder, &graph_is_destroyed));
+          InitializeOpenClApi(&graph, &builder, &graph_is_destroyed,
+                              serialized_kernel_cache, serialized_model_cache,
+                              hold_model_cache));
     } else if (experimental_flags & TFLITE_GPU_EXPERIMENTAL_FLAGS_GL_ONLY) {
       RETURN_IF_ERROR(InitializeOpenGlApi(&graph, &builder));
     } else {
       // By default, we try CL first & fall back to GL if that fails.
       absl::Status status =
-          InitializeOpenClApi(&graph, &builder, &graph_is_destroyed);
+          InitializeOpenClApi(&graph, &builder, &graph_is_destroyed,
+                              serialized_kernel_cache, serialized_model_cache,
+                              hold_model_cache);
       if (!status.ok()) {
         TF_LITE_KERNEL_LOG(context, std::string(status.message()).c_str());
         TF_LITE_KERNEL_LOG(context, "Falling back to OpenGL");
@@ -224,6 +231,22 @@ class DelegateKernel {
     return absl::OkStatus();
   }
 
+  std::vector<uint8_t> GetSerializedModelCache() const {
+      return cl_model_serialized_data_;
+  }
+
+  std::vector<uint8_t> GetSerializedKernelCache() const {
+    if (cl_environment_) {
+      return cl_environment_->GetSerializedBinaryCache();
+    }
+    return {};
+  }
+
+  void DeleteSerializedModelCache() {
+      cl_model_serialized_data_.clear();
+      cl_model_serialized_data_.shrink_to_fit();
+  }
+
  private:
   absl::Status SetInputsAndOutputs(TfLiteContext* context) {
     for (int i = 0; i < input_indices_.size(); ++i) {
@@ -283,9 +306,13 @@ class DelegateKernel {
 
   absl::Status InitializeOpenClApi(GraphFloat32* graph,
                                    std::unique_ptr<InferenceBuilder>* builder,
-                                   bool* graph_is_destroyed) {
+                                   bool* graph_is_destroyed,
+                                   absl::Span<const uint8_t> kernel_cache,
+                                   absl::Span<const uint8_t> model_cache,
+                                   bool hold_model_cache = false) {
     *graph_is_destroyed = false;
     cl::InferenceEnvironmentOptions env_options;
+    env_options.serialized_binary_cache = kernel_cache;
     cl::InferenceEnvironmentProperties properties;
     RETURN_IF_ERROR(cl::NewInferenceEnvironment(env_options, &cl_environment_,
                                                 &properties));
@@ -307,10 +334,19 @@ class DelegateKernel {
     }
     options.usage = ToUsage(delegate_options.inference_preference);
     *graph_is_destroyed = true;
-    RETURN_IF_ERROR(cl_environment_->NewInferenceBuilder(
-        options, std::move(*graph), builder));
-    TFLITE_LOG_PROD_ONCE(tflite::TFLITE_LOG_INFO,
-                         "Initialized OpenCL-based API.");
+    if (!model_cache.empty()) {
+        RETURN_IF_ERROR(cl_environment_->NewInferenceBuilder(model_cache, builder));
+    } else if (hold_model_cache) {
+      RETURN_IF_ERROR(cl_environment_->BuildSerializedModel(
+              options, std::move(*graph), &cl_model_serialized_data_));
+      RETURN_IF_ERROR(cl_environment_->NewInferenceBuilder(
+              cl_model_serialized_data_, builder));
+    } else {
+       RETURN_IF_ERROR(cl_environment_->NewInferenceBuilder(
+               options, std::move(*graph), builder));
+    }
+
+    TFLITE_LOG_PROD_ONCE(tflite::TFLITE_LOG_INFO, "Initialized OpenCL-based API.");
     return absl::OkStatus();
   }
 
@@ -345,6 +381,7 @@ class DelegateKernel {
   std::unique_ptr<gl::InferenceEnvironment> gl_environment_;
 #endif
   std::unique_ptr<InferenceRunner> runner_;
+  std::vector<uint8_t> cl_model_serialized_data_;
   std::vector<int64_t> input_indices_;
   std::vector<int64_t> output_indices_;
   // Whenever quantized inference is enabled, this maps the tensor index of each
@@ -374,7 +411,14 @@ TfLiteStatus DelegatePrepare(TfLiteContext* context, TfLiteDelegate* delegate) {
         // for whatever reason forbids that.
         auto gpu_delegate_kernel =
             absl::make_unique<DelegateKernel>(gpu_delegate);
-        const auto status = gpu_delegate_kernel->Prepare(context, params);
+        auto opts = gpu_delegate->options();
+        auto kernel_cache = absl::MakeSpan(opts.serialized_kernel_cache,
+                                           opts.serialized_kernel_cache_size);
+        auto model_cache = absl::MakeSpan(opts.serialized_model_cache,
+                                          opts.serialized_model_cache_size);
+        const auto status = gpu_delegate_kernel->Prepare(context, params,
+                                                         kernel_cache, model_cache,
+                                                         opts.hold_model_cache);
         if (!status.ok()) {
           TF_LITE_KERNEL_LOG(context, "TfLiteGpuDelegate Init: %s",
                              std::string(status.message()).c_str());
@@ -464,3 +508,61 @@ TfLiteDelegate* TfLiteGpuDelegateV2Create(
 void TfLiteGpuDelegateV2Delete(TfLiteDelegate* delegate) {
   delete tflite::gpu::GetDelegate(delegate);
 }
+
+TFL_CAPI_EXPORT uint8_t* TfLiteGpuDelegateGetKernelCache(const TfLiteNode& node,
+                                                         size_t* size) {
+    auto delegate = reinterpret_cast<tflite::gpu::DelegateKernel*>(node.user_data);
+    if (delegate) {
+        auto cache = delegate->GetSerializedKernelCache();
+        if (cache.size() > 0) {
+            auto data = new uint8_t [cache.size()];
+            *size = cache.size();
+            memcpy(data, cache.data(), cache.size());
+            return data;
+        }
+    }
+}
+
+TFL_CAPI_EXPORT uint8_t* TfLiteGpuDelegateGetModelCache(const TfLiteNode& node,
+                                                        size_t* size,
+                                                        bool delete_after) {
+    auto delegate = reinterpret_cast<tflite::gpu::DelegateKernel*>(node.user_data);
+    if (delegate) {
+        auto cache = delegate->GetSerializedModelCache();
+        if (cache.size() > 0) {
+            auto data = new uint8_t [cache.size()];
+            *size = cache.size();
+            memcpy(data, cache.data(), cache.size());
+            if (delete_after) {
+                delegate->DeleteSerializedModelCache();
+            }
+            return data;
+        }
+    }
+    return nullptr;
+}
+
+TFL_CAPI_EXPORT bool TfLiteGpuDelegateOpenCLSupported() {
+    return tflite::gpu::cl::OpenCLSupported();
+}
+
+TFL_CAPI_EXPORT const char* TfLiteGpuDelegateOpenCLVersion(size_t* size) {
+    *size = 0;
+    tflite::gpu::cl::CLDevice gpu;
+    if (tflite::gpu::cl::CreateDefaultGPUDevice(&gpu).ok()) {
+        auto error = tflite::gpu::cl::clGetDeviceInfo(gpu.id(), CL_DRIVER_VERSION, 0, nullptr, size);
+        if (error != CL_SUCCESS) {
+            return nullptr;
+        }
+        auto str = new char[*size];
+        str[*size - 1] = 0;
+        error = tflite::gpu::cl::clGetDeviceInfo(gpu.id(), CL_DRIVER_VERSION, *size, str, nullptr);
+        if (error != CL_SUCCESS) {
+            delete[] str;
+            return nullptr;
+        }
+        (*size)--;
+        return str;
+    }
+    return nullptr;
+}
diff --git tensorflow/lite/delegates/gpu/delegate.h tensorflow/lite/delegates/gpu/delegate.h
index 40a06bb4384..e2943018fa9 100644
--- tensorflow/lite/delegates/gpu/delegate.h
+++ tensorflow/lite/delegates/gpu/delegate.h
@@ -101,6 +101,30 @@ typedef struct {
   // This limits the maximum number of partitions to be delegated. By default,
   // it's set to 1 in TfLiteGpuDelegateOptionsV2Default().
   int32_t max_delegated_partitions;
+
+  // The buffer for the kernel cache to load into OpenCL if supported and
+  // selected. If the kernel cache is valid, the start up time will avoid
+  // compiling kernels that can be found in the cache. If the cache is invalid,
+  // it will be ignored.
+  uint8_t *serialized_kernel_cache;
+
+  // The buffer size for the kernel cache
+  int64_t serialized_kernel_cache_size;
+
+  // Tells the system to cache the model cache to be queried after the
+  // delegate is setup. The model cache is usually fairly large compared to
+  // the kernel cache, so holding the cache in memory each run should be avoided.
+  bool hold_model_cache;
+
+  // The buffer for the model cache to load into OpenCL if supported and
+  // selected. If the model cache is valid, the start up time will avoid
+  // setting up the OpenCL environment and skip tuning for device.
+  // If the cache is invalid, it will fail to create the model and could
+  // crash.
+  uint8_t *serialized_model_cache;
+
+  // The buffer size for the kernel cache
+  int64_t serialized_model_cache_size;
 } TfLiteGpuDelegateOptionsV2;
 
 // Populates TfLiteGpuDelegateOptionsV2 as follows:
@@ -126,6 +150,27 @@ TFL_CAPI_EXPORT TfLiteDelegate* TfLiteGpuDelegateV2Create(
 // Destroys a delegate created with `TfLiteGpuDelegateV2Create` call.
 TFL_CAPI_EXPORT void TfLiteGpuDelegateV2Delete(TfLiteDelegate* delegate);
 
+// Gets the kernel cache from the specific node if it is a kernel. Usually gained
+// after successfully attaching delegate to graph on the last node. The data must
+// be deleted after use as ownership is passed on.
+TFL_CAPI_EXPORT uint8_t* TfLiteGpuDelegateGetKernelCache(const TfLiteNode& node,
+                                                         size_t* size);
+
+// Gets the model cache from the specific node if it is a kernel. Usually gained
+// after successfully attaching delegate to graph on the last node. Model cache
+// is only available if specified from the TfLiteGpuDelegateOptionsV2. The data
+// must be deleted after use as ownership is passed on.
+TFL_CAPI_EXPORT uint8_t* TfLiteGpuDelegateGetModelCache(const TfLiteNode& node,
+                                                        size_t* size,
+                                                        bool delete_after = false);
+
+// Determine if OpenCL is supported by device with TensorFlow
+TFL_CAPI_EXPORT bool TfLiteGpuDelegateOpenCLSupported();
+
+// Gets the full OpenCL driver version. The string must be deleted after use as
+// ownership is passed on.
+TFL_CAPI_EXPORT const char* TfLiteGpuDelegateOpenCLVersion(size_t* size);
+
 #ifdef __cplusplus
 }
 #endif  // __cplusplus
diff --git tensorflow/lite/delegates/gpu/gl/kernels/resize.cc tensorflow/lite/delegates/gpu/gl/kernels/resize.cc
index 64259f48e7c..baffda4bb56 100644
--- tensorflow/lite/delegates/gpu/gl/kernels/resize.cc
+++ tensorflow/lite/delegates/gpu/gl/kernels/resize.cc
@@ -36,11 +36,13 @@ class Resize : public NodeShader {
   absl::Status GenerateCode(const GenerationContext& ctx,
                             GeneratedCode* generated_code) const final {
     const auto& attr = absl::any_cast<const Resize2DAttributes&>(ctx.op_attr);
-
+#if 0
     if (ctx.input_shapes[0][2] > ctx.output_shapes[0][2] ||
         ctx.input_shapes[0][1] > ctx.output_shapes[0][1]) {
       return absl::InvalidArgumentError("Output size is less than input size.");
     }
+#endif
+
     if (ctx.output_shapes[0][2] != attr.new_shape.w ||
         ctx.output_shapes[0][1] != attr.new_shape.h) {
       return absl::InvalidArgumentError(
