--- tensorflow/tensorflow.bzl	2022-09-21 14:46:15.000000000 -0700
+++ tensorflow/tensorflow.bzl	2022-09-21 14:49:18.000000000 -0700
@@ -2514,6 +2514,7 @@
     return {
         # Force all targets in the subgraph to build on the local machine.
         "//command_line_option:modify_execution_info": ".*=+no-remote-exec",
+        "//command_line_option:platforms": [ "@local_execution_config_platform//:platform" ],
     }
 
 # A transition that forces all targets in the subgraph to be built locally.
@@ -2522,6 +2523,7 @@
     inputs = [],
     outputs = [
         "//command_line_option:modify_execution_info",
+        "//command_line_option:platforms",
     ],
 )
 
