diff -ruN a/tensorflow/lite/build_def.bzl b/tensorflow/lite/build_def.bzl
--- tensorflow/lite/build_def.bzl	2023-09-27 23:49:33.000000000 +0000
+++ tensorflow/lite/build_def.bzl	2024-05-10 09:40:24.023829065 +0000
@@ -279,7 +279,7 @@
       out: name of the output JSON file.
     """
 
-    flatc = "@flatbuffers//:flatc"
+    flatc = "@the8thwall//bzl/redist/flatc:flatc"
     schema = "//tensorflow/lite/schema:schema.fbs"
     native.genrule(
         name = name,
@@ -302,7 +302,7 @@
       out: name of the output flatbuffer file.
     """
 
-    flatc = "@flatbuffers//:flatc"
+    flatc = "@the8thwall//bzl/redist/flatc:flatc"
     schema = "//tensorflow/lite/schema:schema_fbs"
     native.genrule(
         name = name,
diff -ruN a/tensorflow/lite/schema/BUILD b/tensorflow/lite/schema/BUILD
--- tensorflow/lite/schema/BUILD	2023-09-27 23:49:33.000000000 +0000
+++ tensorflow/lite/schema/BUILD	2024-05-10 09:40:24.023829065 +0000
@@ -40,7 +40,7 @@
         "schema_v1.fbs",
         "schema_v2.fbs",
         "schema_v3.fbs",
-        "@flatbuffers//:flatc",
+        "@the8thwall//bzl/redist/flatc:flatc",
     ],
     srcs_version = "PY3",
     deps = [
@@ -150,7 +150,7 @@
     deps = [
         "//tensorflow/core/platform",
         "@com_google_googletest//:gtest_main",
-        "@flatbuffers//:flatc_library",
+        "@the8thwall//bzl/redist/flatc:flatc_library",
     ],
 )
 
diff -ruN a/third_party/flatbuffers/build_defs.bzl b/third_party/flatbuffers/build_defs.bzl
--- third_party/flatbuffers/build_defs.bzl	2023-09-27 23:49:33.000000000 +0000
+++ third_party/flatbuffers/build_defs.bzl	2024-05-10 09:40:24.023829065 +0000
@@ -2,7 +2,7 @@
 
 load("@build_bazel_rules_android//android:rules.bzl", "android_library")
 
-flatc_path = "@flatbuffers//:flatc"
+flatc_path = "@the8thwall//bzl/redist/flatc:flatc"
 zip_files = "//tensorflow/lite/tools:zip_files"
 
 DEFAULT_INCLUDE_PATHS = [
@@ -360,7 +360,7 @@
             mandatory = False,
         ),
         "_flatc": attr.label(
-            default = Label("@flatbuffers//:flatc"),
+            default = Label("@the8thwall//bzl/redist/flatc:flatc"),
             executable = True,
             cfg = "host",
         ),
