From 31545a4323809323da7649121ebddc4891fa4653 Mon Sep 17 00:00:00 2001
From: Filippo Trimoldi <filippotrimoldi@nianticlabs.com>
Date: Wed, 1 Mar 2023 15:03:15 +0000
Subject: [PATCH] Upgrading some bazel apple rule dependency for compatibility
 with bazel 6.0.0

---
 tensorflow/workspace2.bzl | 23 +++++++++--------------
 1 file changed, 9 insertions(+), 14 deletions(-)

diff --git a/tensorflow/workspace2.bzl b/tensorflow/workspace2.bzl
index e35e4da267f..6a68208a2de 100644
--- tensorflow/workspace2.bzl
+++ tensorflow/workspace2.bzl
@@ -813,32 +813,27 @@ def _tf_repositories():
 
     # Apple and Swift rules.
     # https://github.com/bazelbuild/rules_apple/releases
+    # Taking version https://github.com/bazelbuild/rules_apple/releases
     tf_http_archive(
         name = "build_bazel_rules_apple",
-        sha256 = "7984c3aafbaa971c42c3507195bcdd68c814efd1b4034a012f3dae29d5694f80",
-        strip_prefix = "rules_apple-942c24380a6cb001d3521c510fdb9b61ee164585",
-        urls = [
-            "https://storage.googleapis.com/mirror.tensorflow.org/github.com/bazelbuild/rules_apple/archive/942c24380a6cb001d3521c510fdb9b61ee164585.tar.gz",
-            "https://github.com/bazelbuild/rules_apple/archive/942c24380a6cb001d3521c510fdb9b61ee164585.tar.gz",
-        ],
+        sha256 = "3e2c7ae0ddd181c4053b6491dad1d01ae29011bc322ca87eea45957c76d3a0c3",
+        urls = tf_mirror_urls("https://github.com/bazelbuild/rules_apple/releases/download/2.1.0/rules_apple.2.1.0.tar.gz"),
     )
 
     # https://github.com/bazelbuild/rules_swift/releases
+    # Taking version https://github.com/bazelbuild/rules_apple/releases/tag/2.1.0
     tf_http_archive(
         name = "build_bazel_rules_swift",
-        sha256 = "12057b7aa904467284eee640de5e33853e51d8e31aae50b3fb25d2823d51c6b8",
-        urls = tf_mirror_urls("https://github.com/bazelbuild/rules_swift/releases/download/1.0.0/rules_swift.1.0.0.tar.gz"),
+        sha256 = "d25a3f11829d321e0afb78b17a06902321c27b83376b31e3481f0869c28e1660",
+        urls = tf_mirror_urls("https://github.com/bazelbuild/rules_swift/releases/download/1.6.0/rules_swift.1.6.0.tar.gz"),
     )
 
     # https://github.com/bazelbuild/apple_support/releases
+    # Taking version https://github.com/bazelbuild/apple_support/releases/tag/1.4.1
     tf_http_archive(
         name = "build_bazel_apple_support",
-        sha256 = "ac0d1d0d231784bffc70f28e74427e6ccaae63cf65ce6039ab52382ab1c16d88",
-        strip_prefix = "apple_support-f7bcd10eebdfb427098fcaa73060fd6dd3f1aaea",
-        urls = [
-            "https://storage.googleapis.com/mirror.tensorflow.org/github.com/bazelbuild/apple_support/archive/f7bcd10eebdfb427098fcaa73060fd6dd3f1aaea.tar.gz",
-            "https://github.com/bazelbuild/apple_support/archive/f7bcd10eebdfb427098fcaa73060fd6dd3f1aaea.tar.gz",
-        ],
+        sha256 = "77a121a0f5d4cd88824429464ad2bfb54bdc8a3bccdb4d31a6c846003a3f5e44",
+        urls = tf_mirror_urls("https://github.com/bazelbuild/apple_support/releases/download/1.4.1/apple_support.1.4.1.tar.gz"),
     )
 
     # https://github.com/apple/swift-protobuf/releases
-- 
2.39.2

