diff --git tensorflow/lite/delegates/gpu/common/task/texture2d_desc.h tensorflow/lite/delegates/gpu/common/task/texture2d_desc.h
index 4178799c6b6..b63bd9a808a 100644
--- tensorflow/lite/delegates/gpu/common/task/texture2d_desc.h
+++ tensorflow/lite/delegates/gpu/common/task/texture2d_desc.h
@@ -28,7 +28,7 @@ struct Texture2DDescriptor : public GPUObjectDescriptor {
   DataType element_type;
   bool normalized = false;   // used with INT data types, if normalized, we read
                              // in kernel float data.
-  DataType normalized_type;  // can be FLOAT32 or FLOAT16, using with normalized
+  DataType normalized_type = DataType::UNKNOWN;  // can be FLOAT32 or FLOAT16, using with normalized
                              // = true
 
   // optional
