diff --git a/node_modules/eslint-plugin-local-rules/index.js b/node_modules/eslint-plugin-local-rules/index.js
index 4b00064..7f277be 100644
--- a/node_modules/eslint-plugin-local-rules/index.js
+++ b/node_modules/eslint-plugin-local-rules/index.js
@@ -3,7 +3,10 @@
 
 var path = require('path');
 
-var rules = requireUp('eslint-local-rules.js', __dirname);
+// NOTE(christoph): Since __dirname might be in a external/npm-eslint/node_modules
+// folder separate from the main path, we fallback to CWD to resolve the entry point.
+var rules = requireUp('eslint-local-rules.js', __dirname) ||
+            requireUp('eslint-local-rules.js', process.cwd());
 
 if (!rules) {
   throw new Error(
