diff --git i/build_defs.bzl w/build_defs.bzl
index b8217a18d..b17b23649 100644
--- i/build_defs.bzl
+++ w/build_defs.bzl
@@ -152,7 +152,6 @@ def xnnpack_cc_library(
             "-Iinclude",
             "-Isrc",
         ] + copts + select({
-            ":linux_k8": gcc_x86_copts,
             ":linux_arm": aarch32_copts,
             ":linux_armeabi": aarch32_copts,
             ":linux_armhf": aarch32_copts,
@@ -195,7 +194,6 @@ def xnnpack_cc_library(
         includes = ["include", "src"] + includes,
         linkstatic = True,
         linkopts = select({
-            ":linux_k8": ["-lpthread"],
             ":linux_arm": ["-lpthread"],
             ":linux_armeabi": ["-lpthread"],
             ":linux_armhf": ["-lpthread"],
