diff --git a/include/spdlog/logger.h b/include/spdlog/logger.h
index a8abae41..cf40de7f 100644
--- a/include/spdlog/logger.h
+++ b/include/spdlog/logger.h
@@ -347,6 +347,8 @@ public:
     // create new logger with same sinks and configuration.
     virtual std::shared_ptr<logger> clone(std::string logger_name);

+    virtual void sink_it_(const details::log_msg &msg);
+    void log_it_(const details::log_msg &log_msg, bool log_enabled, bool traceback_enabled);
 protected:
     std::string name_;
     std::vector<sink_ptr> sinks_;
@@ -407,8 +409,6 @@ protected:

     // log the given message (if the given log level is high enough),
     // and save backtrace (if backtrace is enabled).
-    void log_it_(const details::log_msg &log_msg, bool log_enabled, bool traceback_enabled);
-    virtual void sink_it_(const details::log_msg &msg);
     virtual void flush_();
     void dump_backtrace_();
     bool should_flush_(const details::log_msg &msg);
